src/OverviewView.js

```diff
@@ -0,0 +1,141 @@
+ * @param {Object} viewState The viewState object.
+ * @param {Object} loader The loader, used for inferring zoom level.
+ * @param {number} detailHeight The height of the detail view.
+ * @param {number} detailWidth The width of the detail view.
+ * @param {number} overviewScale The scale of this viewport relative to the detail.
+ * @param {number} margin The margin to be offset from the the corner of the other viewport.
+ * @param {string} overviewLocation The location of the viewport - one of "bottom-right", "top-right", "top-left", "bottom-left."
+ * */
+export default class OverviewView extends VivView {
+  constructor({
+    viewState,
+    loader,
+    detailHeight,
+    detailWidth,
+    overviewScale,
+    margin = 25,
+    overviewLocation = 'bottom-right'
+  }) {
+    super({ viewState });
+    this.margin = margin;
+    this.loader = loader;
+    this.overviewLocation = overviewLocation;
+    this.detailHeight = detailHeight;
+    this.detailWidth = detailWidth;
+    this._setHeightWidthScale({ detailWidth, overviewScale });
+    this._setXY();
+  }
+
+  _setHeightWidthScale({ detailWidth, overviewScale }) {
+    const { loader } = this;
+    const { numLevels } = loader;
+    const rasterSize = loader.getRasterSize({
+      z: 0
+    });
+    const heightWidthRatio = rasterSize.height / rasterSize.width;
+    this.width = detailWidth * overviewScale;
+    this.height = this.width * heightWidthRatio;
+    this.overviewScale = (2 ** (numLevels - 1) / rasterSize.width) * this.width;
+    this._imageWidth = rasterSize.width;
+    this._imageHeight = rasterSize.height;
+  }
+
+  /**
+   * Set the x and y of the current view.
+   */
+  _setXY() {
+    const {
+      height,
+      width,
+      margin,
+      overviewLocation,
+      detailWidth,
+      detailHeight
+    } = this;
+    switch (overviewLocation) {
+      case 'bottom-right': {
+        this.x = detailWidth - width - margin;
+        this.y = detailHeight - height - margin;
+        break;
+      }
+      case 'top-right': {
+        this.x = detailWidth - width - margin;
+        this.y = margin;
+        break;
+      }
+      case 'top-left': {
+        this.x = margin;
+        this.y = margin;
+        break;
+      }
+      case 'bottom-left': {
+        this.x = margin;
+        this.y = detailHeight - height - margin;
+        break;
+      }
+      default: {
+        throw new Error(
+          `overviewLocation prop needs to be one of ['bottom-right', 'top-right', 'top-left', 'bottom-left']`
+        );
+      }
+    }
+  }
+
+  getDeckGlView() {
+    const { x, y, id, height, width } = this;
+    return new OrthographicView({
+      id,
+      controller: false,
+      height,
+      width,
+      x,
+      y,
+      clear: true
+    });
+  }
+
+  getViewState(viewState) {
+    // Scale the view as the overviewScale changes with screen resizing - basically, do not react to any view state changes.
```

Comment:
One thing that is occurring to me now (and sorry for bringing this up) is that with using `scale`, `detailWidth`, and `detailHeight` means that if the user adjusts the size of their screen, the size of the detail view will grow and shrink. This is kind of awkward, especially if I have a detail view full screen with a overview view, if I go to half screen the overview would become tiny. Initially I thought having this resize be dynamic would be a nice feature, but I'm almost more inclined to say we should force the user to provide an absolute width (and use an absolute default as well).

Reply:
I am a little confused at what you're asking.  It seems you're saying that it's awkward for the `DetailView` to change size, and then by extension, for the `OverviewView` to change with it.  But the detail view doesn't have to resize - that is in `App.js` and is not a built-in part of `Viv`, and those size parameters are absolute.  We could just not pass in dynamically the changing height and width to `VivViewer`.  Or are you saying that `DetailView` should change size, and `OverviewView` should not?  Or, should `DetailView` not change size, and `OverviewView` remain pinned to whatever corner is in the bottom of the screen?  Sorry for all the questions but I want to understand what you're saying and also get this right.